observability-test/observability.ts

```diff
@@ -0,0 +1,450 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const assert = require('assert');
+const {
+  AlwaysOffSampler,
+  AlwaysOnSampler,
+  NodeTracerProvider,
+  InMemorySpanExporter,
+} = require('@opentelemetry/sdk-trace-node');
+const {SpanStatusCode, TracerProvider} = require('@opentelemetry/api');
+const {SimpleSpanProcessor} = require('@opentelemetry/sdk-trace-base');
+const {
+  SPAN_NAMESPACE_PREFIX,
+  getActiveOrNoopSpan,
+  setSpanError,
+  setSpanErrorAndException,
+  startTrace,
+} = require('../src/instrument');
+const {
+  SEMATTRS_DB_NAME,
+  SEMATTRS_DB_SQL_TABLE,
+  SEMATTRS_DB_STATEMENT,
+  SEMATTRS_DB_SYSTEM,
+  SEMATTRS_EXCEPTION_MESSAGE,
+} = require('@opentelemetry/semantic-conventions');
+
+const {disableContextAndManager, setGlobalContextManager} = require('./helper');
+
+const {
+  AsyncHooksContextManager,
+} = require('@opentelemetry/context-async-hooks');
+
+describe('startTrace', () => {
+  const globalExporter = new InMemorySpanExporter();
+  const sampler = new AlwaysOnSampler();
+
+  const globalProvider = new NodeTracerProvider({
+    sampler: sampler,
+    exporter: globalExporter,
+  });
+  globalProvider.addSpanProcessor(new SimpleSpanProcessor(globalExporter));
+  globalProvider.register();
+
+  const contextManager = new AsyncHooksContextManager();
+  setGlobalContextManager(contextManager);
+
+  afterEach(() => {
+    globalExporter.forceFlush();
+  });
+
+  after(async () => {
+    globalExporter.forceFlush();
+    await globalProvider.shutdown();
+    disableContextAndManager(contextManager);
+  });
+
+  it('with TracerProvider in global configuration', () => {
+    startTrace('mySpan', {}, async span => {
+      await new Promise((resolve, reject) => setTimeout(resolve, 400));
+      span.end();
+
+      const spans = globalExporter.getFinishedSpans();
+      assert.strictEqual(
+        spans.length,
+        1,
+        'Exactly 1 span must have been exported'
+      );
+      const span0 = spans[0];
+
+      assert.equal(
+        span0.name,
+        SPAN_NAMESPACE_PREFIX + '/mySpan',
+        'name mismatch'
+      );
+    });
+  });
+
+  it('with TracerProvider in options, skips using global TracerProvider', () => {
+    const overridingExporter = new InMemorySpanExporter();
+    const overridingProvider = new NodeTracerProvider({
+      sampler: sampler,
+      exporter: overridingExporter,
+    });
+    overridingProvider.addSpanProcessor(
+      new SimpleSpanProcessor(overridingExporter)
+    );
+    overridingProvider.register();
```

Comment:
If we are registering this here, how would we make sure if the tests are using the injected provider or the registered one ? 
Shouldn't we remove this line and just inject the tracerprovider ?

Reply:
Roger that, I thought about it last evening too, done and thank you @surbhigarg92!